pkgs/build-support/rust/default.nix

```
@@ -95,14 +98,13 @@ stdenv.mkDerivation (args // {
     ''}
     EOF
 
-    unset cargoDepsCopy
     export RUST_LOG=${logLevel}
-  '' + stdenv.lib.optionalString verifyCargoDeps ''
-    if ! diff source/Cargo.lock $cargoDeps/Cargo.lock ; then
+  '' + stdenv.lib.optionalString (!legacyCargoFetcher) ''
```

```suggestion
  '' + stdenv.lib.optionalString (cargoDeps.passthru ? checker) cargoDeps.passthru.checker 
```
This is just an idea I want to mention, it doesn't need to be acted upon.

If that code was moved to the fetcher, it would allow to pass `cargoDeps` to the buildRustPackage and make this whole thing more composable. Right now the arguments to buildRustPackage contain both arguments for it and the fetcher which makes it quite wide. By separating the cargoDeps it would allow for example to change the compression of the deps in a local development scenario.

It's just an idea. It also has downsides since some of the arguments like the `src` are currently shared.

---

Hmm, I'll ponder it over -- but in the meantime, while looking at this area of the code and while prepping some work for the next step in the mass migration where we set `legacyCargoFetcher = false;` by default, I noticed that this block of code gets entered into even when `cargoSha256 == "unset";`, e.g. building `cargo` itself.  I've updated accordingly to check that we actually have a `cargoSha256` before validating it now.